cadquery/occ_impl/exporters/dxf.py

```diff
@@ -1,154 +1,328 @@
-from ...cq import Workplane, Plane, Face
-from ...units import RAD2DEG
-from ..shapes import Edge
-from .utils import toCompound
-
-from OCP.gp import gp_Dir
-from OCP.GeomConvert import GeomConvert
+"""DXF export utilities."""
 
-from typing import Optional, Literal
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
 
 import ezdxf
+from ezdxf import units, zoom
+from ezdxf.entities import factory
+from OCP.GeomConvert import GeomConvert
+from OCP.gp import gp_Dir
 
-CURVE_TOLERANCE = 1e-9
-
-
-def _dxf_line(e: Edge, msp: ezdxf.layouts.Modelspace, plane: Plane):
-
-    msp.add_line(
-        e.startPoint().toTuple(), e.endPoint().toTuple(),
-    )
-
-
-def _dxf_circle(e: Edge, msp: ezdxf.layouts.Modelspace, plane: Plane):
-
-    geom = e._geomAdaptor()
-    circ = geom.Circle()
-
-    r = circ.Radius()
-    c = circ.Location()
-
-    c_dy = circ.YAxis().Direction()
-    c_dz = circ.Axis().Direction()
-
-    dy = gp_Dir(0, 1, 0)
-
-    phi = c_dy.AngleWithRef(dy, c_dz)
-
-    if c_dz.XYZ().Z() > 0:
-        a1 = RAD2DEG * (geom.FirstParameter() - phi)
-        a2 = RAD2DEG * (geom.LastParameter() - phi)
-    else:
-        a1 = -RAD2DEG * (geom.LastParameter() - phi) + 180
-        a2 = -RAD2DEG * (geom.FirstParameter() - phi) + 180
-
-    if e.IsClosed():
-        msp.add_circle((c.X(), c.Y(), c.Z()), r)
-    else:
-        msp.add_arc((c.X(), c.Y(), c.Z()), r, a1, a2)
-
-
-def _dxf_ellipse(e: Edge, msp: ezdxf.layouts.Modelspace, plane: Plane):
-
-    geom = e._geomAdaptor()
-    ellipse = geom.Ellipse()
-
-    r1 = ellipse.MinorRadius()
-    r2 = ellipse.MajorRadius()
+from ...cq import Face, Plane, Workplane
+from ...units import RAD2DEG
+from ..shapes import Edge
+from .utils import toCompound
 
-    c = ellipse.Location()
-    xdir = ellipse.XAxis().Direction()
-    xax = r2 * xdir.XYZ()
+ApproxOptions = Literal["spline", "arc"]
+DxfEntityAttributes = Tuple[str, Dict[str, Any]]
```

Comment:
Could you type the first `str` using `Literal['SPLINE','ARC','CIRCLE','ELLIPSE']`